src/background/messaging/rpc-methods/stx-deploy-contract.ts

```diff
@@ -0,0 +1,67 @@
+import { ClarityVersion } from '@stacks/transactions';
+import { createUnsecuredToken } from 'jsontokens';
+
+import {
+  RpcErrorCode,
+  type StxDeployContractRequest,
+  type StxDeployContractRequestParams,
+  stxDeployContractRequestParamsSchema,
+} from '@leather.io/rpc';
+import { TransactionTypes } from '@leather.io/stacks';
+
+import { RouteUrls } from '@shared/route-urls';
+import { RpcErrorMessage } from '@shared/rpc/methods/validation.utils';
+import { makeRpcErrorResponse } from '@shared/rpc/rpc-methods';
+
+import {
+  RequestParams,
+  getStxDefaultMessageParamsToTransactionRequest,
+  listenForPopupClose,
+  makeSearchParamsWithDefaults,
+  triggerRequestWindowOpen,
+  validateRequestParams,
+} from '../messaging-utils';
+import { trackRpcRequestSuccess } from '../rpc-message-handler';
+
+function getMessageParamsToTransactionRequest(params: StxDeployContractRequestParams) {
+  const defaultParams = getStxDefaultMessageParamsToTransactionRequest(params);
+
+  return {
+    txType: TransactionTypes.ContractDeploy,
+    contractName: params.name,
+    codeBody: params.clarityCode,
+    clarityVersion: params.clarityVersion ?? ClarityVersion.Clarity3,
+    ...defaultParams,
+  };
+}
+
+export async function rpcStxDeployContract(
```

Comment:
We can prob combine these into one shared function? There are 3 that are identical just with diff params.